### PR TITLE
Fixes for make compatible for new data model and improvements.

### PR DESCRIPTION
**Pull request by sadilchamishka**
## Purpose

- [x] The new data model for organization management consists of new tables instead of already available tables in the IS. Therefore the existing queries have to be updated.
 - The authorization logic related sql query
 - List organizations based on the permission the requested user assigned for organizations in the given tenant

- [x] Added a query param for the get organization by id endpoint to retieve the assigned permissions for the user for the requested organization along with the response. 

The basic permissions assigned for the users are

- /permission/admin/manage/identity/organizationmgt/create
- /permission/admin/manage/identity/organizationmgt/update
- /permission/admin/manage/identity/organizationmgt/view
- /permission/admin/manage/identity/organizationmgt/delete

Sample Request 
```{{base_url}}/t/{{tenant}}/api/identity/organization-mgt/v1.0/organizations/{{org-id}}?showChildren=true&includePermissions=true```

Sample Response
```
{
    "id": "4e128465-1ae3-4156-b78d-8172e27af601",
    "name": "1",
    "description": "this is a construction company",
    "status": "ACTIVE",
    "created": "2022-06-14T07:36:53Z",
    "lastModified": "2022-06-14T07:36:53Z",
    "type": "STRUCTURAL",
    "parent": {
        "id": "root4a8d-113f-4211-a0d5-efe36b082211",
        "ref": "/t/carbon.super/api/identity/organization-mgt/v1.0/organizations/root4a8d-113f-4211-a0d5-efe36b082211"
    },
    "attributes": [
        {
            "key": "Country",
            "value": "Sri Lanka"
        },
        {
            "key": "Language",
            "value": "Sinhala"
        }
    ],
    "permissions": [
        "/permission/admin/manage/identity/organizationmgt/create",
        "/permission/admin/manage/identity/organizationmgt/view",
        "/permission/admin/manage/identity/organizationmgt/update",
        "/permission/admin/manage/identity/organizationmgt/delete"
    ]
}

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java [hunk 1]
```diff
@@ -71,10 +71,12 @@ public interface OrganizationManager {
      *
      * @param organizationId The organization ID.
      * @param showChildren   Whether the child organizations should be retrieved.
+     * @param includePermissions   Whether the permissions attached for organizations should be retrieved.
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java:74]: not formatted properly

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/model/Organization.java [hunk 2]
```diff
@@ -148,4 +149,12 @@ public void setTenantId(int tenantId) {
...
         this.tenantId = tenantId;
     }
+
+    public List<String> getPermissions() {
+        return permissions;
+    }
+
+    public void setPermissions(List<String> permissions) {
+        this.permissions = permissions;
+    }
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/model/Organization.java:159]: not formatted properly

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java [hunk 3]
```diff
@@ -192,4 +192,14 @@ boolean isAttributeExistByKey(String organizationId, String attributeKey)
      * @throws OrganizationManagementServerException The server exception thrown when retrieving the organization type.
      */
     String getOrganizationType(String organizationId) throws OrganizationManagementServerException;
+
+    /**
+     * Get list of permissions assigned to user for the organization.
+     * @param organizationId
+     * @param userId
+     * @return list of resource ids of permissions
+     * @throws OrganizationManagementServerException
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java:201]: describe the params

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManager.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/39/commits/3dc5d13f9a52b0cc8024ac2617ce5f557bdbd8be

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/model/Organization.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/model/Organization.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/39/commits/3dc5d13f9a52b0cc8024ac2617ce5f557bdbd8be

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java (hunk 3 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/OrganizationManagementDAO.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/39/commits/3dc5d13f9a52b0cc8024ac2617ce5f557bdbd8be

File: components/org.wso2.carbon.identity.organization.management.endpoint/src/main/resources/org.wso2.carbon.identity.organization.management.yaml [hunk 4]
```diff
@@ -312,6 +313,15 @@ components:
       schema:
         type: boolean
         default: false
+    includePermissionsQueryParam:
+      in: query
+      name: includePermissions
+      required: false
+      description:
+        Returns the organization details along with permissions assigned for the requested user for this organization.
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.endpoint/src/main/resources/org.wso2.carbon.identity.organization.management.yaml:321]:
```suggestion
        Returns the organization details along with permissions assigned for the requested user in this organization.
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 5]
```diff
@@ -341,6 +351,9 @@ public enum ErrorMessages {
                 "Server encountered an error when linking the application: %s to shared application: %s."),
         ERROR_CODE_ERROR_RESOLVING_ENTERPRISE_IDP_LOGIN("65058", "Unable to resolve the enterpriseIDP shared app." +
                 "login", "Server encountered an error when resolving enterpriseIDP login for application: %s."),
+        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_PERMISSIONS("65059", "Unable to retrieve organizations permissions",
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:354]:
```suggestion
        ERROR_CODE_ERROR_RETRIEVING_ORGANIZATION_PERMISSIONS("65059", "Unable to retrieve organizations permissions.",
```

File: components/org.wso2.carbon.identity.organization.management.endpoint/src/main/resources/org.wso2.carbon.identity.organization.management.yaml (hunk 4 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.endpoint/src/main/resources/org.wso2.carbon.identity.organization.management.yaml]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/39/commits/d5962156b1cf33dfcd077ccb877f2462ad47fa27

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 5 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/39/commits/9075b595a99ab74938ff638626eef283ad166535

**Review by dewniMW — APPROVED** [review]

**Comment by dewniMW** [comment]
@sadilchamishka as discussed let's add unit tests in a separate PR

**Pull request merged**